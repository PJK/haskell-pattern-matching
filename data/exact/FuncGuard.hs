module FuncGuard where

isPrimeOrSmall :: Int -> Bool
isPrimeOrSmall x
    | isPrime x && x < 10 = True
    | not (isPrime x)        = False

