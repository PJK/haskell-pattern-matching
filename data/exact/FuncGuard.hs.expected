{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            [
                "isPrimeOrSmall",
                {
                    "tag": "NonExhaustive",
                    "contents": [
                        [
                            [
                                {
                                    "tag": "VariablePattern",
                                    "contents": "~a"
                                }
                            ],
                            "Constraints:\n  ~f == False\n  ~f == not (isPrime x)\n  ~c == False\n  ~c == isPrime x && ~a < 10\n\nSatisfiable. Model:\n  ~f          = False :: Bool\n  (isPrime x) =  True :: Bool\n  ~c          = False :: Bool\n  isPrime x   = False :: Bool\n  ~a          =     9 :: Integer\n"
                        ]
                    ]
                }
            ]
        ],
        [
            [
                "isPrimeOrSmall",
                [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ]
                ]
            ]
        ]
    ]
}
